### PR TITLE
Staking fixes

### PR DESCRIPTION
**Pull request by dorin-iancu**
- APR limit applied globally instead of per user
- move staking SC tests in its own folder

### CODE REVIEW

**Review by claudiulataretu — CHANGES_REQUESTED** [review]

File: dex/farm-staking/src/custom_rewards.rs [hunk 1]
```diff
@@ -15,11 +15,19 @@ pub trait CustomRewardsModule:
         let last_reward_nonce = self.last_reward_block_nonce().get();
 
         if current_block_nonce > last_reward_nonce {
-            let extra_rewards =
+            let extra_rewards_unbounded =
                 self.calculate_per_block_rewards(current_block_nonce, last_reward_nonce);
+
+            let farm_token_supply = self.farm_token_supply().get();
+            let extra_rewards_apr_bounded_per_block =
+                self.get_amount_apr_bounded(&farm_token_supply);
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply dex/farm-staking/src/custom_rewards.rs:23]: +1

File: dex/farm-staking/src/custom_rewards.rs [hunk 2]
```diff
@@ -135,22 +143,31 @@ pub trait CustomRewardsModule:
         amount: &BigUint,
         current_reward_per_share: &BigUint,
         initial_reward_per_share: &BigUint,
-        last_claim_block: u64,
+        _last_claim_block: u64,
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply dex/farm-staking/src/custom_rewards.rs:146]: This has to be deleted. It does not do what it says

File: dex/farm-staking/src/custom_rewards.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (sasurobert) [reply dex/farm-staking/src/custom_rewards.rs:23]: This is per block. So it should be ok

[the remark this answers is not in the document]
Comment (sasurobert) [reply dex/farm-staking/src/custom_rewards.rs:23]: Why don't we limit both local/user APR and global as well?

**Review by sasurobert — COMMENTED** [review]
Limit both local and global

**Review by claudiulataretu — APPROVED** [review]

File: dex/farm-staking/src/custom_rewards.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (claudiulataretu) [reply dex/farm-staking/src/custom_rewards.rs:23]: It works as it is. I had to double check for myself

**Review by sasurobert — APPROVED** [review]

**Review by catalinnnn — APPROVED** [review]

**Review by claudiulataretu — APPROVED** [review]

**Pull request merged**